homework/src/index.js

```diff
@@ -31,17 +32,151 @@
   }
 
   function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      'class': 'header'
+    });
+    const container = createAndAppend('div', root, {
+      'class': 'container'
+    });
+    createAndAppend('p', header, {
+      'html': 'HYF Repositories'
+    });
+    const select = createAndAppend('select', header, {
+      'id': 'list'
+    });
+
+
+    const repositoriesInfoSec = createAndAppend('section', container, {
+      'class': 'repos-info-sec box'
+    });
+
+    const table = createAndAppend('table', repositoriesInfoSec, {
+      'id': 'info-table'
+    });
+    const repoNameLine = createAndAppend('tr', table);
+    createAndAppend('td', repoNameLine, {
+      'html': 'Repository :',
+      'class': 'label'
+    });
+    const repoName = createAndAppend('td', repoNameLine, {
+      'id': 'repoName'
+    });
+    const repoNameLink = createAndAppend('a', repoName, {
+      'target': '_blank',
+    });
+
+    const descriptionLine = createAndAppend('tr', table);
+    const descriptionLabel = createAndAppend('td', descriptionLine, {
+      'html': 'Description :',
+      'class': 'label'
+    });
+    const descriptionValue = createAndAppend('td', descriptionLine);
+    const forkLine = createAndAppend('tr', table);
+    createAndAppend('td', forkLine, {
+      'html': 'Forks :',
+      'class': 'label'
+    });
+    const forkValue = createAndAppend('td', forkLine);
+
+    const updateLine = createAndAppend('tr', table);
+    createAndAppend('td', updateLine, {
+      'html': 'Updated :',
+      'class': 'label'
+    });
+    const updateValue = createAndAppend('td', updateLine);
+    const contributorsSec = createAndAppend('section', container, {
+      'class': 'contributors-sec box'
+    });
+    createAndAppend('p', contributorsSec, {
+      html: 'Contributions',
+      'class': 'contributions'
+    });
+    const ul = createAndAppend('ul', contributorsSec);
+
+
     fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
+        createAndAppend('div', container, {
+          html: err.message,
+          class: 'alert-error'
+        });
       } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        data.sort((a, b) => a.name.localeCompare(b.name, 'fr', {
+          ignorePunctuation: true
+        }));
+
+        data.forEach((element, i) => {
+          createAndAppend('option', select, {
+            'html': element.name,
+            'value': i
+          });
+        });
+
+        showRepositories(0, data);
+
+        const contributorsInfo = data[0].contributors_url;
+        fetchJSON(contributorsInfo, (err, contributorData) => {
```

Comment:
Avoid (meaningless) suffixes such as `data`. A better name for `contributorData` would be `contributors`. This plural form suggest that the data is actually an array.

The name `contributorsInfo` is misleading. It is actually a URL so might better be called `contributorsUrl`.

Reply:
done 👍